DialURL should ignore opaque url

Opaque url is currently ignored by `Dial`:

```
$ cat main.go
package main

import "github.com/gomodule/redigo/redis"

func main() {
	// c, err := redis.DialURL("redis:http://localhost:6379")
	c, err := redis.Dial("tcp", "redis:http://localhost:6379")
	if err != nil {
		panic(err)
	}
	defer c.Close()

	resp, err := redis.String(c.Do("PING"))
	if err != nil {
		panic(err)
	}
	println(resp)
}
$ go run main.go
panic: dial tcp: address redis:http://localhost:6379: too many colons in address

goroutine 1 [running]:
main.main()
	/home/cuonglm/bin/go/main.go:9 +0x1e6
exit status 2
```

But `DialURL` accepts it.

For consistency, we should reject opaque url for `DialURL`, too.

The current behavior also leads to confusion, using `redis:http://localhost:6379` works but `redis:http://localhost:6379/1` returns invalid database error.

--- a/redis/conn.go
+++ b/redis/conn.go
@@ -265,6 +265,10 @@ func DialURL(rawurl string, options ...DialOption) (Conn, error) {
 		return nil, fmt.Errorf("invalid redis URL scheme: %s", u.Scheme)
 	}
 
+	if u.Opaque != "" {
+		return nil, fmt.Errorf("invalid redis URL, url is opaque: %s", rawurl)
+	}
+
 	// As per the IANA draft spec, the host defaults to localhost and
 	// the port defaults to 6379.
 	host, port, err := net.SplitHostPort(u.Host)

--- a/redis/conn_test.go
+++ b/redis/conn_test.go
@@ -553,6 +553,10 @@ var dialErrors = []struct {
 		"redis://localhost:6379/abc123",
 		"invalid database: abc123",
 	},
+	{
+		"redis:foo//localhost:6379",
+		"invalid redis URL, url is opaque: redis:foo//localhost:6379",
+	},
 }
 
 func TestDialURLErrors(t *testing.T) {

